driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -145,6 +146,19 @@ public void initChannel(final SocketChannel ch) throws Exception {
             public void operationComplete(final ChannelFuture future) throws Exception {
                 if (future.isSuccess()) {
                     channel = channelFuture.channel();
+                    channel.closeFuture().addListener(new ChannelFutureListener() {
+                        @Override
+                        public void operationComplete(final ChannelFuture f2) throws Exception {
+                            PendingReader pending = null;
```

Comment:
I think this can be simplified to:

    handleReadResponse(null, new IOException("The connection to the server was closed"))